### PR TITLE
Update cluster-api-provider-vsphere-*

### PR DESCRIPTION
**Pull request by ncdc**
Move cluster-api-provider-vsphere-* teams from kubernetes-sigs config
file into the sig-cluster-lifecycle config file.

Add detiber, ncdc, vincepri, yastij to maintainers team.

### CODE REVIEW

**Comment by ncdc** [comment]
@timothysc does this file shuffling make sense?

@akutz are these changes ok with you?

**Comment by ncdc** [comment]
cc @detiber @vincepri

**Comment by justaugustus** [comment]
Shuffle makes sense!
/lgtm
/approve
/hold for SIG CL approval

**Comment by ncdc** [comment]
FYI @yastij

**Comment by justaugustus** [comment]
@ncdc -- For clarity, can you split this into two commits?
One for the move and then a second for the additional team members?

**Comment by ncdc** [comment]
@justaugustus updated!

**Comment by justaugustus** [comment]
@ncdc -- Thanks! Feel free to remove the hold once a CL approver gives the lgtm

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml [hunk 1]
```diff
@@ -175,6 +175,32 @@ teams:
     - jichenjc
     - sbueringer
     privacy: closed
+  cluster-api-provider-vsphere-admins:
+    description: Admin access to the cluster-api-provider-vsphere repo
+    maintainers:
+    - spiffxp
```
Comment (timothysc) [inline config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml]: I think we should put one of the core maintainers here.  In this case @yastij

Comment (ncdc) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml]: Updated

**Pull request merged**